rotkehlchen/db/history_events.py

```diff
@@ -200,44 +225,75 @@ def get_customized_event_identifiers(
 
         return [x[0] for x in cursor]
 
-    def get_history_event_by_identifier(self, identifier: int) -> Optional[HistoryBaseEntry]:
+    def get_evm_event_by_identifier(self, identifier: int) -> Optional['EvmEvent']:
         """Returns the history event with the given identifier"""
         with self.db.conn.read_ctx() as cursor:
-            cursor.execute('SELECT * FROM history_events WHERE identifier=?', (identifier,))
-            entry = cursor.fetchone()
-            if entry is None:
+            event_data = cursor.execute(
+                f'SELECT {HISTORY_BASE_ENTRY_FIELDS}, {EVM_EVENT_FIELDS} {EVM_EVENT_JOIN} WHERE history_events.identifier=? AND entry_type=?',  # noqa: E501
+                (identifier, HistoryBaseEntryType.EVM_EVENT),
+            ).fetchone()
+            if event_data is None:
+                log.debug(f'Didnt find event with identifier {identifier}')
                 return None
 
         try:
-            deserialized = HistoryBaseEntry.deserialize_from_db(entry)
+            deserialized = EvmEvent.deserialize_from_db(event_data[1:])
         except (DeserializationError, UnknownAsset) as e:
-            log.debug(f'Failed to deserialize history event {entry} due to {str(e)}')
+            log.debug(f'Failed to deserialize evm event {event_data} due to {str(e)}')
             return None
 
         return deserialized
 
+    @overload
     def get_history_events(
             self,
             cursor: 'DBCursor',
             filter_query: HistoryEventFilterQuery,
             has_premium: bool,
     ) -> list[HistoryBaseEntry]:
+        ...
+
+    @overload
+    def get_history_events(  # type: ignore[misc]  # getting from mypy "Overloaded function signature 2 will never be matched: signature 1's parameter type(s) are the same or broader"  # noqa: E501
```

Comment:
Just write that as a comment in the PR not in the code. I will look at this, I guess since this is what I commented in the other parts too.